package.json

```diff
@@ -18,11 +18,12 @@
   "files": [
     "/dist/index.js",
     "/dist/index.mjs",
-    "/dist/index.d.ts"
+    "/dist/index.d.ts",
+    "/dist/index.d.mts"
   ],
   "scripts": {
     "test": "tsc -noEmit -p tsconfig-test.json && jest --useStderr --runInBand --detectOpenHandles",
-    "build": "npm run lint && tsup src/index.ts --format cjs,esm --dts --clean --platform neutral --minify",
+    "build": "npm run lint && tsup src/index.ts --format cjs,esm --dts --clean --platform neutral --minify && cp ./dist/index.d.ts ./dist/index.d.mts",
```

Comment:
Simply duplicate declaration file with `d.mts` extension. This will make typescript compiler happy because for `.mjs` module types will be visible.